src/eduid_common/api/decorators.py

```diff
@@ -89,6 +110,10 @@ def __call__(self, f):
         @wraps(f)
         def decorated_function(*args, **kwargs):
             ret = f(*args, **kwargs)
+
+            if isinstance(ret, Response):  # No need to Marshal again, someone else already did that
+                return ret
+
```

Comment:
I'm finding here that not all responses are flask.Response. For example, a redirect, that comes directly from werkzeug, produces a werkzeug.wrappers.Response type response.

So I think the check should be for `isinstance(ret, webob.Response)`.

In my version I was also turning a 500 response into a 200 with the error in the json body. I'm not sure this is really needed, I seem to remember I added it to fix a failing test, so for the moment we better forget it, and keep it in mind in case errors appear.